lib/publiccloud/azure.pm

```diff
@@ -25,15 +25,35 @@ has image_gallery => 'test_image_gallery';
 has lease_id => undef;
 has storage_region => 'westeurope';
 
-my $default_sku = 'gen2';
-
 sub init {
     my ($self) = @_;
     $self->SUPER::init();
     $self->provider_client(publiccloud::azure_client->new());
     $self->provider_client->init();
 }
 
+=head2 az_arch
+    Provides architecture tag x64 or Arm64 compatible with Azure syntax, based on '*ARCH' job settings
+=cut
+
+sub az_arch {
+    return (get_var('PUBLIC_CLOUD_ARCH', get_required_var('ARCH'))) eq 'x86_64' ? 'x64' : 'Arm64';
+}
+
+=head2 az_sku
+    Centralized sku definition and default value, based on PUBLIC_CLOUD_AZURE_SKU
+
+    Call: az_sku(<default>);
+        when PUBLIC_CLOUD_AZURE_SKU undefined, the default from input is used:
+        when <default> is undefined too, hardcoded string used; i.e. az_sku(), default = 'gen2'; 
+=cut
+
+sub az_sku {
+    my ($self, $default) = @_;
```

Comment:
Why would you use the `$default` parameter?
Also `$self` is not used at all.
```suggestion
```

Reply:
default is not used, but is placed there to allow to `change` default if/when needed.
Moreover the `self` I _had to add_ because in a previous test it failed because missing.

Reply:
Much clearly, adding an input parameter to the az_sku routine I have to skip the `self`, that also is present in all the azure.pm routines using input parmeters

Reply:
I think `$default` will be never used Maurizio :( :)

Reply:
Infact, if one let input empty, the behavior is transparent, as before. But the code is always subject to reviews and updates, we cannot tell `nothing will happen...`  :eyeglasses:

Reply:
Having unused variables is a red flag for any linter.  I don't know why it wasn't flagged by CI.  Let's keep the `$self` and drop the `$default` like this: `my $self = shift;`

When the variable is needed it can be added, but having unused variables is a big no-no.

Reply:
Not this the case, we dealing with input parameters and the $default is used inside the routine. The $self contains the class instance coming from routine call:no linter claim because it is up to the programmer how the routine is called with related input paramerers, depending on needs.